OADP-5570: Improving DataMover Support (#87746)
openshift/openshift-docs: backup_and_restore/application_backup_and_restore/installing/about-oadp-data-mover.adoc

{oadp-short} supports CSI snapshots on the following:

* {odf-full}
* Any other cloud storage provider with the Container Storage Interface (CSI) driver that supports the Kubernetes Volume Snapshot API

// ** suggest removing the admonition and replacing with a clear section **
//
// [IMPORTANT]
// ====
// The {oadp-short} built-in Data Mover, which was introduced in {oadp-short} 1.3 as a Technology Preview, is now fully supported for both containerized and virtual machine workloads.
// ====

[id="oadp-data-mover-support_{context}"]
== Data Mover support

Backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 using the Data Mover feature cannot be restored using {oadp-short} 1.3 and later. Therefore, it is not supported.

> {oadp-short} Data Mover backups taken with {oadp-short} 1.3 and later are supported for restores in {oadp-short} 1.3 and later.

Author: ```suggestion
The Data Mover backups taken with {oadp-short} 1.3 can be restored using {oadp-short} 1.3, 1.4, and later. This is supported.

In {oadp-short} 1.1 and 1.2, Data Mover does not support backups and restores. Data Mover in  {oadp-short} 1.1 and 1.2 remains a Technology Preview so it is therefore not supported. 

Data Mover backups that were taken using {oadp-short} 1.1 and 1.2 cannot be restored using {oadp-short} 1.3 and later.
```

toc::[]

{oadp-first} includes a built-in Data Mover that you can use to move Container Storage Interface (CSI) volume snapshots to a remote object store. The built-in Data Mover allows you to restore stateful applications from the remote object store if a failure, accidental deletion, or corruption of the cluster occurs. It uses xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-about-kopia.adoc#oadp-about-kopia[Kopia] as the uploader mechanism to read the snapshot data and write to the unified repository.

{oadp-short} supports CSI snapshots on the following:

* {odf-full}
* Any other cloud storage provider with the Container Storage Interface (CSI) driver that supports the Kubernetes Volume Snapshot API

// ** suggest removing the admonition and replacing with a clear section **
//
// [IMPORTANT]
// ====

> // The {oadp-short} built-in Data Mover, which was introduced in {oadp-short} 1.3 as a Technology Preview, is now fully supported for both containerized and virtual machine workloads.

Reviewer: this is correct: The {oadp-short} built-in Data Mover, which was introduced in {oadp-short} 1.3 as a Technology Preview, is now fully supported for both containerized and virtual machine workloads.

Author: moved to being intro in the support section

// ====

[id="oadp-data-mover-support_{context}"]
== Data Mover support

> The {oadp-short} Data Mover feature was added to {oadp-short} 1.3 as a technology preview. As a technology preview, it is not supported.

Reviewer: this is NOT correct: The {oadp-short} Data Mover feature was added to {oadp-short} 1.3 as a technology preview. As a technology preview, it is not supported.  
Please remove

> The {oadp-short} Data Mover feature was added to {oadp-short} 1.3 as a technology preview. As a technology preview, it is not supported.

Author: ```suggestion
```

Backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 using the Data Mover feature cannot be restored using {oadp-short} 1.3 and later. Therefore, it is not supported.

> {oadp-short} Data Mover backups taken with {oadp-short} 1.3 and later are supported for restores in {oadp-short} 1.3 and later.

Author: @weshayutin - not sure if this is what you meant, please could you clarify and i will have another crack at it

Reviewer: Data Mover backups taken with {oadp-short} 1.3 are supported for restores in {oadp-short} 1.3, {oadp-short} 1.4 and future releases.  

I'm explicitly stating that 1.3 backups can be restored by 1.4, 1.5 etc..  The context is that some customers have  a requirement to hold on to backups for years.  These old backups are supported by future releases of oadp for restores.   

While also stating 1.1, 1.2 DM backup/restores were NEVER supported, always tech preview and can NOT be restored by current or future supported versions of OADP.

toc::[]

{oadp-first} includes a built-in Data Mover that you can use to move Container Storage Interface (CSI) volume snapshots to a remote object store. The built-in Data Mover allows you to restore stateful applications from the remote object store if a failure, accidental deletion, or corruption of the cluster occurs. It uses xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-about-kopia.adoc#oadp-about-kopia[Kopia] as the uploader mechanism to read the snapshot data and write to the unified repository.

{oadp-short} supports CSI snapshots on the following:

* {odf-full}
* Any other cloud storage provider with the Container Storage Interface (CSI) driver that supports the Kubernetes Volume Snapshot API

// ** suggest removing the admonition and replacing with a clear section **
//
// [IMPORTANT]
// ====
// The {oadp-short} built-in Data Mover, which was introduced in {oadp-short} 1.3 as a Technology Preview, is now fully supported for both containerized and virtual machine workloads.
// ====

[id="oadp-data-mover-support_{context}"]
== Data Mover support

> 

Author: ```suggestion
The {oadp-short} built-in Data Mover, which was introduced in {oadp-short} 1.3 as a Technology Preview, is now fully supported for both containerized and virtual machine workloads.

```

toc::[]

{oadp-first} includes a built-in Data Mover that you can use to move Container Storage Interface (CSI) volume snapshots to a remote object store. The built-in Data Mover allows you to restore stateful applications from the remote object store if a failure, accidental deletion, or corruption of the cluster occurs. It uses xref:../../../backup_and_restore/application_backup_and_restore/backing_up_and_restoring/oadp-about-kopia.adoc#oadp-about-kopia[Kopia] as the uploader mechanism to read the snapshot data and write to the unified repository.

{oadp-short} supports CSI snapshots on the following:

* {odf-full}
* Any other cloud storage provider with the Container Storage Interface (CSI) driver that supports the Kubernetes Volume Snapshot API

// ** suggest removing the admonition and replacing with a clear section **
//
// [IMPORTANT]
// ====
// The {oadp-short} built-in Data Mover, which was introduced in {oadp-short} 1.3 as a Technology Preview, is now fully supported for both containerized and virtual machine workloads.
// ====

[id="oadp-data-mover-support_{context}"]
== Data Mover support

> 

Reviewer 2: ```suggestion

.Supported

* The {oadp-short} built-in Data Mover, which was introduced in {oadp-short} 1.3 as a Technology Preview, is now fully supported for both containerized and virtual machine workloads.

* The Data Mover backups taken with {oadp-short} 1.3 can be restored using {oadp-short} 1.3, 1.4, and later. This is supported.

.Not supported

* Backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 using the Data Mover feature cannot be restored using {oadp-short} 1.3 and later. Therefore, it is not supported.

* In {oadp-short} 1.1 and 1.2, Data Mover does not support backups and restores. Data Mover in {oadp-short} 1.1 and 1.2 remains a Technology Preview so it is therefore not supported.

* Data Mover backups that were taken using {oadp-short} 1.1 and 1.2 cannot be restored using {oadp-short} 1.3 and later.
```

I really like having a separate section to express the level of support. Here is my suggestion to make it clearer. I have only separated it out into `Supported` and `Not supported`. Feel free to keep it or toss it.

The {oadp-short} built-in Data Mover, which was introduced in {oadp-short} 1.3 as a Technology Preview, is now fully supported for both containerized and virtual machine workloads.

Backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 using the Data Mover feature cannot be restored using {oadp-short} 1.3 and later. Therefore, it is not supported.

The Data Mover backups taken with {oadp-short} 1.3 can be restored using {oadp-short} 1.3, 1.4, and later. This is supported.

> In {oadp-short} 1.1 and 1.2, Data Mover does not support backups and restores. Data Mover in  {oadp-short} 1.1 and 1.2 remains a Technology Preview so it is therefore not supported. 

Reviewer: I think we can remove this line (32).  Or  replace it with 
OADP-1.1 and OADP-1.2 are no longer supported.  The DataMover feature in  OADP-1.1 or OADP-1.2 was tech preview and never supported.  DataMover backups taken with OADP-1.1 or OADP-1.2 can not be restored on later versions of OADP.  ( Your call )

> In {oadp-short} 1.1 and 1.2, Data Mover does not support backups and restores. Data Mover in  {oadp-short} 1.1 and 1.2 remains a Technology Preview so it is therefore not supported. 

Author: ```suggestion
{oadp-short} 1.1 and {oadp-short} 1.2 are no longer supported.  The DataMover feature in {oadp-short} 1.1 or {oadp-short} 1.2 was a Technology Preview and never supported.  DataMover backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 cannot be restored on later versions of {oadp-short}. 
```

> {oadp-short} 1.1 and {oadp-short} 1.2 are no longer supported.  The DataMover feature in {oadp-short} 1.1 or {oadp-short} 1.2 was a Technology Preview and never supported.  DataMover backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 cannot be restored on later versions of {oadp-short}. 

Author: ```suggestion
{oadp-short} 1.1 and {oadp-short} 1.2 are no longer supported. The DataMover feature in {oadp-short} 1.1 or {oadp-short} 1.2 was a Technology Preview and never supported.  DataMover backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 cannot be restored on later versions of {oadp-short}. 
```

{oadp-short} supports CSI snapshots on the following:

* {odf-full}
* Any other cloud storage provider with the Container Storage Interface (CSI) driver that supports the Kubernetes Volume Snapshot API

// ** suggest removing the admonition and replacing with a clear section **
//
// [IMPORTANT]
// ====
// The {oadp-short} built-in Data Mover, which was introduced in {oadp-short} 1.3 as a Technology Preview, is now fully supported for both containerized and virtual machine workloads.
// ====

[id="oadp-data-mover-support_{context}"]
== Data Mover support

The {oadp-short} built-in Data Mover, which was introduced in {oadp-short} 1.3 as a Technology Preview, is now fully supported for both containerized and virtual machine workloads.

.Supported

The Data Mover backups taken with {oadp-short} 1.3 can be restored using {oadp-short} 1.3, 1.4, and later. This is supported.

.Not supported

Backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 using the Data Mover feature cannot be restored using {oadp-short} 1.3 and later. Therefore, it is not supported.

> {oadp-short} 1.1 and {oadp-short} 1.2 are no longer supported. The DataMover feature in {oadp-short} 1.1 or {oadp-short} 1.2 was a Technology Preview and never supported. DataMover backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 cannot be restored on later versions of {oadp-short}.

Reviewer 3: NIT: Technology Preview and **_was_** never supported

Author: Nothing to see here... LOL

> {oadp-short} 1.1 and {oadp-short} 1.2 are no longer supported. The DataMover feature in {oadp-short} 1.1 or {oadp-short} 1.2 was a Technology Preview and never supported. DataMover backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 cannot be restored on later versions of {oadp-short}.

Author: ```suggestion
{oadp-short} 1.1 and {oadp-short} 1.2 are no longer supported. The DataMover feature in {oadp-short} 1.1 or {oadp-short} 1.2 was a Technology Preview and was never supported. DataMover backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 cannot be restored on later versions of {oadp-short}.
```

{oadp-short} supports CSI snapshots on the following:

* {odf-full}
* Any other cloud storage provider with the Container Storage Interface (CSI) driver that supports the Kubernetes Volume Snapshot API

// ** suggest removing the admonition and replacing with a clear section **
//
// [IMPORTANT]
// ====
// The {oadp-short} built-in Data Mover, which was introduced in {oadp-short} 1.3 as a Technology Preview, is now fully supported for both containerized and virtual machine workloads.
// ====

[id="oadp-data-mover-support_{context}"]
== Data Mover support

The {oadp-short} built-in Data Mover, which was introduced in {oadp-short} 1.3 as a Technology Preview, is now fully supported for both containerized and virtual machine workloads.

.Supported

Data Mover backups that were taken using {oadp-short} 1.1 and 1.2 cannot be restored using {oadp-short} 1.3 and later.

.Not supported

Backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 using the Data Mover feature cannot be restored using {oadp-short} 1.3 and later. Therefore, it is not supported.

The Data Mover backups taken with {oadp-short} 1.3 can be restored using {oadp-short} 1.3, 1.4, and later. This is supported.

> {oadp-short} 1.1 and {oadp-short} 1.2 are no longer supported. The DataMover feature in {oadp-short} 1.1 or {oadp-short} 1.2 was a Technology Preview and never supported.  DataMover backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 cannot be restored on later versions of {oadp-short}.

Author: ```suggestion
{oadp-short} 1.1 and {oadp-short} 1.2 are no longer supported. The DataMover feature in {oadp-short} 1.1 or {oadp-short} 1.2 was a Technology Preview and never supported. DataMover backups taken with {oadp-short} 1.1 or {oadp-short} 1.2 cannot be restored on later versions of {oadp-short}.
```
